xarray/backends/h5netcdf_.py

```diff
@@ -55,43 +86,63 @@ def __init__(self, filename, mode='r', format=None, group=None,
         opener = functools.partial(_open_h5netcdf_group, filename, mode=mode,
                                    group=group)
         self.ds = opener()
+        self._isopen = True
         self.format = format
         self._opener = opener
         self._filename = filename
         self._mode = mode
         super(H5NetCDFStore, self).__init__(writer)
 
+    @contextlib.contextmanager
+    def ensure_open(self, autoclose=True):
```

Comment:
I think I see what you are saying-- essentially we need to make sure that this action only happens if we specify, on loading a dataset via `open_mfdataset` that `autoclose=True`.  Otherwise, we never close a file, correct?